src/DTS/eBaySDK/Types/BaseType.php

```diff
@@ -459,7 +569,8 @@ private static function ensurePropertyType($class, $name, $value)
         $expectedType = $info['type'];
         $actualType = self::getActualType($value);
 
-        if ($expectedType !== $actualType && 'array' !== $actualType) {
+        if (($expectedType !== $actualType && 'array' !== $actualType) 
+                && ('double' === $expectedType &&  !is_numeric($value))) {
```

Comment:
This changes causes the tests to fail with the following errors.

DTS\eBaySDK\Types\Test\ConstructTest::testSettingInvalidPropertyTypeViaCtor
Failed asserting that exception of type "\DTS\eBaySDK\Exceptions\InvalidPropertyTypeException" is thrown.

DTS\eBaySDK\Types\Test\SimpleClassTest::testSettingPropertyWithAnInvalidType
Failed asserting that exception of type "\DTS\eBaySDK\Exceptions\InvalidPropertyTypeException" is thrown.